app/controllers/preauthorize_transactions_controller.rb

```diff
@@ -260,96 +239,13 @@ def booked
-      booking:  false,
-      quantity: quantity,
-      listing_price: vprms[:listing][:price],
-      localized_unit_type: translate_unit_from_listing(vprms[:listing]),
-      localized_selector_label: translate_selector_label_from_listing(vprms[:listing]),
-      subtotal: (quantity > 1) ? vprms[:subtotal] : nil,
-      total: vprms[:total_price]
-    })
-
-    render "listing_conversations/preauthorize", locals: {
-      preauthorize_form: PreauthorizeMessageForm.new,
-      braintree_client_side_encryption_key: braintree_settings[:braintree_client_side_encryption_key],
-      braintree_form: BraintreeForm.new,
-      listing: vprms[:listing],
-      quantity: quantity,
-      author: query_person_entity(vprms[:listing][:author_id]),
-      action_button_label: vprms[:action_button_label],
-      expiration_period: MarketplaceService::Transaction::Entity.authorization_expiration_period(vprms[:payment_type]),
-      form_action: preauthorized_payment_path(person_id: @current_user.id, listing_id: vprms[:listing][:id]),
-      price_break_down_locals: price_break_down_locals
-    }
-  end
-
-  def preauthorized
-    conversation_params = params[:listing_conversation]
-
-    if @current_community.transaction_agreement_in_use? && conversation_params[:contract_agreed] != "1"
-      flash[:error] = t("error_messages.transaction_agreement.required_error")
-      return redirect_to action: :preauthorize
-    end
-
-    preauthorize_form = PreauthorizeMessageForm.new(conversation_params.merge({
-      listing_id: @listing.id
-    }))
-
-    if preauthorize_form.valid?
-      braintree_form = BraintreeForm.new(params[:braintree_payment])
-      quantity = TransactionViewUtils.parse_quantity(preauthorize_form.quantity)
-
-      transaction_response = TransactionService::Transaction.create({
-          transaction: {
-            community_id: @current_community.id,
-            listing_id: @listing.id,
-            listing_title: @listing.title,
-            starter_id: @current_user.id,
-            listing_author_id: @listing.author.id,
-            unit_type: @listing.unit_type,
-            unit_price: @listing.price,
-            unit_tr_key: @listing.unit_tr_key,
-            listing_quantity: quantity,
-            content: preauthorize_form.content,
-            payment_gateway: :braintree,
-            payment_process: :preauthorize,
-          },
-          gateway_fields: braintree_form.to_hash
-        })
-
-      unless transaction_response[:success]
-        flash[:error] = "An error occured while trying to create a new transaction: #{transaction_response[:error_msg]}"
-        return redirect_to action: :preauthorize
-      end
-
-      transaction_id = transaction_response[:data][:transaction][:id]
-
-      redirect_to person_transaction_path(:person_id => @current_user.id, :id => transaction_id)
+    if (transaction_response[:data][:gateway_fields][:redirect_url])
```

Comment:
If transaction_reponse is a Result then shouldn't this be done with on_success instead?